tests/components/modbus/conftest.py

```diff
@@ -67,8 +70,11 @@ async def base_test(
         "homeassistant.components.modbus.modbus.ModbusUdpClient", return_value=mock_sync
     ):
 
-        # Setup inputs for the sensor
-        read_result = ReadResult(register_words)
+        read_result = (
+            ReadResult(register_words)
+            if register_words
+            else ModbusException("Emulate Modbus read exception")
```

Comment:
That is a change I really like. That means adding exception testing to all test files.